src/test/rustdoc/process-termination.rs

```diff
@@ -0,0 +1,24 @@
+// compile-flags:--test
+
+/// A check of using various process termination strategies
+///
+/// # Examples
+///
+/// ```rust
+/// assert!(true); // this returns `()`, all is well
+/// ```
+///
+/// You can also simply return `Ok(())`, but you'll need to disambiguate the
+/// type using turbofish, because we cannot infer the type:
+///
+/// ```rust
+/// Ok::<(), &'static str>(())
+/// ```
+///
+/// You can err with anything that implements `Debug`:
+///
+/// ```rust,should_panic
+/// Err("This is returned from `main`, leading to panic")?;
```

Comment:
I find this use of `should_panic` confusing, since that's _not_ how `#[test]`s work: https://github.com/rust-lang/rust/issues/49909 https://github.com/rust-lang/rust/issues/48854#issuecomment-375353118 https://github.com/rust-lang/rust/pull/49911

I think that `Err` paths should look for `Err`s directly, and that `should_panic` tests should _only_ pass if the code block raised a panic.

Reply:
Ok, but how should I test this particular function then? I could replace the `should_panic` with a `no_run`, but that would fail to catch a change that swallows the panic.